packages/runtime/runtime-definitions/src/dataStoreContext.ts

```diff
@@ -610,6 +613,21 @@ export interface IFluidDataStoreContext extends IFluidParentContext {
 	 * and its children with the GC details from the previous summary.
 	 */
 	getBaseGCDetails(): Promise<IGarbageCollectionDetailsBase>;
+
+	/**
+	 * This function creates a detached child datastore synchronously.
+	 * In order for this function to succeed:
+	 * 1. This datastore's factory must also be a IFluidDataStoreRegistry,
+	 * 2. This datastore's registry must include the same instance as the provided child factory.
+	 * 3. The datastore's registry must synchronously provide the child factory.
+	 * These invariant's ensure that the child datastore can also be created by a remote client
+	 * running the same code as this client.
+	 *
+	 * @param childFactory - The factory of the datastore to be created.
+	 */
+	createChildDataStoreSync?<T extends IFluidDataStoreFactory>(
```

Comment:
I don't think I understand what the motivating scenario is though, so I'd love to learn more about that.  This comment is purely taking this PR at face value, so if I'm missing some special context let me know.

At a very high level, I'd question whether a sync factory pattern is moving in a good direction for our overall instantiation flows.

One pain point of Aqueduct (and therefore almost all of our current data store examples) is that the async startup flows are deferred to the data store instance (i.e. async `initializingFirstTime`, async `hasInitialized`).  This leaves the instance in a prolonged "half-initialized" state which is prone to errors that don't account for certain invariants being violated.

Fluid basically mandates async somewhere in the construction flow though, due to retrieving handles, getChannel, etc.), so we can't go fully sync all the way to full initialization for most scenarios.

An alternative approach (ditching Aqueduct) would be to embrace an async factory, and move all async initialization up to the factory such that by the time we call the instance constructor, we are ready to be fully sync (and the instance is fully initialized in the constructor).

I've explored this pattern in https://github.com/microsoft/FluidFramework/blob/main/examples/utils/migration-tools/src/migrationTool.ts if you'd like an example.  Fully admit there's some personal preference here, but I think this aligns better with DI principles, simplifies the instance class, and strengthens the instance class's invariants.

Reply:
scriptor has a case where they want to create a child datastore in response to user input, so there is no opportunity to do async actions.

i'd not confuse this with all instantiation, which include load, load must always be async. This matches how dds work. 

We can't split the sync part and the async part here. the whole things need to be sync, as we need a net new datastore synchronously that ready to be used, and have its handle stores in the same js turn, to not break on user input.

Reply:
I see, thank you - that makes more sense.